src/js/legend.js

```diff
@@ -0,0 +1,198 @@
+        model: {
+            // dataSet accepts:
+            // 1. an array of objects. Must contain "id", "value" and "label" variables.
+            // Example: [{id: string, value: number, label: string} ... ]
+            dataSet: []
+        },
+        legendOptions: {
+            width: 300,
+            height: 300,
+            colors: null, // An array of colors for the legend generated for corresponding values of model.dataSet
+            sort: true // Whether or not to sort the data by values when creating the legend
+        },
+        styles: {
+            legend: "gpii-ca-pieChart-legend",
+            table: "gpii-ca-pieChart-legend-table",
+            row: "gpii-ca-pieChart-legend-table-row",
+            legendColorCell: "gpii-ca-pieChart-legend-table-row-legend-color-cell",
+            labelCell: "gpii-ca-pieChart-legend-table-row-legend-label-cell",
+            valueCell: "gpii-ca-pieChart-legend-table-row-legend-value-cell"
+
+        },
+        selectors: {
+            legend: ".gpiic-ca-pieChart-legend",
+            table: ".gpii-ca-pieChart-legend-table",
+            row: ".gpii-ca-pieChart-legend-table-row",
+            legendColorCell: ".gpii-ca-pieChart-legend-table-row-legend-color-cell",
+            labelCell: ".gpii-ca-pieChart-legend-table-row-legend-label-cell",
+            valueCell: ".gpii-ca-pieChart-legend-table-row-legend-value-cell"
+        },
+        events: {
+            onLegendCreated: null  // Fire when the legend is created. Ready to register D3 DOM event listeners
+        },
+        listeners: {
+            "onCreate.create": {
+                funcName: "gpii.chartAuthoring.pieChart.legend.create",
+                args: ["{that}"]
+            }
+        },
+        modelListeners: {
+            dataSet: {
+                funcName: "{that}.draw",
+                excludeSource: "init"
+            }
+        },
+        invokers: {
+            draw: {
+                funcName: "gpii.chartAuthoring.pieChart.legend.draw",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    // Takes the dataSet array and the color array, and returns a consolidated object array to ease sorting and other operations while keeping colors "correct"
+
+    gpii.chartAuthoring.pieChart.legend.consolidateDataAndColors = function(dataSet, colors) {
+        var c = [];
+        for(var i=0; i<dataSet.length; i++) {
+            var d = {
+                id: dataSet[i].id,
+                label: dataSet[i].label,
+                value: dataSet[i].value,
+                color: colors(i)
+            };
+            c.push(d);
+        }
+        return c;
+    };
+
+    gpii.chartAuthoring.pieChart.legend.draw = function(that) {
+
+        var table = that.table,
+            l = that.options.legendOptions,
+            dataSet = gpii.chartAuthoring.pieChart.legend.consolidateDataAndColors(that.model.dataSet,that.color),
```

Comment:
This consolidation should use model relay. An example: https://github.com/fluid-project/infusion/blob/master/tests/framework-tests/core/js/DataBindingTests.js#L947-L955